src/Controller/Controller.php

```diff
@@ -243,6 +244,11 @@ public function __construct(Request $request = null, Response $response = null,
             $this->name = substr($name, 0, -10);
         }
 
+        if ($componentRegistry !== null) {
+            $this->_components = $componentRegistry;
+            $this->_components->setController($this);
+        }
+
```

Comment:
It's impossible because we have to set out custom registry before `initialize()` method. According to this docs: http://book.cakephp.org/3.0/en/controllers/components.html#configuring-components